### ISSUE TITLE
CompletableFuture is not unwrapped in controller advice with problem-spring-web

### ISSUE THREAD

**Issue by Blackdread** [zalando/problem-spring-web#389]
## Description
When using CompletableFuture as return type for Controllers, exceptions are not handled correctly in @ControllerAdvice class.

For example, both won't be called:
```
fun handle(ex: IllegalArgumentException, request: NativeWebRequest): ResponseEntity<Problem> {
}
fun handle(ex: BusinessException, request: NativeWebRequest): ResponseEntity<Problem> {
}
```
but this one will be:
```
@ExceptionHandler
    fun handle(ex: CompletionException, request: NativeWebRequest): ResponseEntity<Problem> {
}
```

Not sure this is intentional and if can be resolved since Spring tried to fix this problem.
See [spring-framework 22476](https://github.com/spring-projects/spring-framework/pull/22476)
and other tickets were created for that.

## Expected Behavior
Methods will IllegalArgumentException or BusinessException should be called instead.

## Actual Behavior

Method with CompletionException is always called.

## Steps to Reproduce

1.
2.
3.
4.

## Context

Spring 2.1.9.RELEASE

## Your Environment

* Version used:
* Link to your project:

**Comment by whiskeysierra** [comment]
This has been fixed in Spring 5.1.7, see https://github.com/spring-projects/spring-framework/pull/22476#issuecomment-547768188

**Comment by Blackdread** [comment]
Ok thanks.
As I am using Spring Boot v2.1.9.RELEASE, I have as dependency Spring v5.1.10.RELEASE 
So it should be fixed but still have error @bclozel
